### DISCUSSION TITLE
(PDF) Page.get_pixmap(). How to render mediabox?

### DISCUSSION THREAD

**Discussion by openandclose** [pymupdf/PyMuPDF#1224] [Q&A] [answered]
Hello, It seems `Page.get_pixmap()` renders `Page.rect`.

(1)
How to render `Page.mediabox`?

I find manually setting cropbox to mediabox (and rotation to 0) does work,
but is it a better way?

(2)
Is it intended and stable behavior?
I find the document is not clear enough about what box is used.

https://pymupdf.readthedocs.io/en/latest/page.html#Page.get_pixmap
https://pymupdf.readthedocs.io/en/latest/functions.html#Page.get_displaylist
https://pymupdf.readthedocs.io/en/latest/displaylist.html#DisplayList.get_pixmap
https://pymupdf.readthedocs.io/en/latest/displaylist.html#DisplayList.rect

**Comment by JorjMcKie** [comment, marked as answer]
> It seems Page.get_pixmap() renders Page.rect.

It does. This is intended, because that is what a PDF viewer would show, too.
If you however do need the mediabox, then your way is certainly possible. There are no stability concerns. If you afterwards intend to do other work with the page based on the previous CropBox / rotation values, simply restore both.

I will try to include this comment in the documentation.
Thank you for bringing this up.

**Reply by openandclose** [reply]
Oh, thank you for the clear answer.
I'm relieved.

**Reply by openandclose** [reply]
I can see the `Mark as answer` button,
but it is unresponsive in Firefox and Chronium.

The same situation as @bingalls
https://github.com/github/feedback/discussions/2903#discussioncomment-1196944

So please treat it as it is (as answered).
